components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIImportUtil.java

```diff
@@ -932,4 +941,84 @@ private static void updateAPIWithCertificate(JsonElement certificate, APIProvide
+                            + SOAPToRESTConstants.SequenceGen.SOAP_TO_REST_OUT_RESOURCE;
+            try {
+                // Import inflow mediation logic
+                Path dir = Paths.get(inFlowFileLocation);
+                InputStream inFlowStream = null;
+                try (DirectoryStream<Path> stream = Files.newDirectoryStream(dir)) {
+                    for (Path file : stream) {
+                        String fileName = file.getFileName().toString();
+                        String method = "";
+                        if (fileName.split(".xml").length != 0) {
+                            method = fileName.split(".xml")[0]
+                                    .substring(file.getFileName().toString().lastIndexOf("_") + 1);
+                        }
+                        inFlowStream = new FileInputStream(file.toFile());
+                        byte[] inSeqData = IOUtils.toByteArray(inFlowStream);
+                        Resource inSeqResource = (Resource) registry.newResource();
+                        inSeqResource.setContent(inSeqData);
+                        inSeqResource.addProperty(SOAPToRESTConstants.METHOD, method);
+                        inSeqResource.setMediaType("text/xml");
+                        registry.put(soapToRestLocationIn + RegistryConstants.PATH_SEPARATOR + file.getFileName(),
+                                inSeqResource);
+                    }
+                } finally {
+                    IOUtils.closeQuietly(inFlowStream);
+                }
+                // Import outflow mediation logic
+                dir = Paths.get(outFlowFileLocation);
+                InputStream outFlowStream = null;
+                try (DirectoryStream<Path> stream = Files.newDirectoryStream(dir)) {
+                    for (Path file : stream) {
+                        String fileName = file.getFileName().toString();
+                        String method = "";
+                        if (fileName.split(".xml").length != 0) {
+                            method = fileName.split(".xml")[0]
+                                    .substring(file.getFileName().toString().lastIndexOf("_") + 1);
+                        }
+                        outFlowStream = new FileInputStream(file.toFile());
+                        byte[] inSeqData = IOUtils.toByteArray(outFlowStream);
+                        Resource inSeqResource = (Resource) registry.newResource();
+                        inSeqResource.setContent(inSeqData);
+                        inSeqResource.addProperty(SOAPToRESTConstants.METHOD, method);
+                        inSeqResource.setMediaType("text/xml");
+                        registry.put(soapToRestLocationOut + RegistryConstants.PATH_SEPARATOR + file.getFileName(),
+                                inSeqResource);
+                    }
+                } finally {
+                    IOUtils.closeQuietly(outFlowStream);
```

Comment:
If it's the same logic being duplicated for in location and out location, better to create a common method and re-use it rather than duplicating the code.

Reply:
resolved by #8332